### PR TITLE
Add ladder to pingbon

### PR DESCRIPTION
**Pull request by MikaelZehnterFrisk**

### CODE REVIEW

File: components/LadderPlayerItem.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React, { useState } from 'react';
+import { trimDecimals } from '../lib/helpers';
+
+const LadderPlayerItem = ({ player }) => {
+  const [values, setValues] = useState({ ...player });
+
+  const { _id, __v, ...rest } = player;
```
Comment (adambergman1) [inline components/LadderPlayerItem.js:7]: This line doesn't do anything, so it might as well be removed 😊

File: components/ReportLadderMatch.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        .filter((player) => (player.ladderPosition != 1))
+        .map((player) => ({
+          value: player._id,
+          label: `(${player.ladderPosition}) ${player.name} `,
+        })),
+    );
+  }, [players]);
+
+  useEffect(() => {
+    if (!challenger || players.length < 0) {
```
Comment (adambergman1) [inline components/ReportLadderMatch.js]: The latter condition: is it intentional to check for a negative length? Wouldn't it be more readable to check whether it has no length, or if it's falsy?

File: components/LadderPlayerItem.js (hunk 1 above)
Comment (adambergman1) [inline components/LadderPlayerItem.js]: This line doesn't do anything, so it might as well be removed 😊

File: components/ReportLadderMatch.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      return
+    }
+    const challengerPosition = players.find((player) => player._id === challenger.value).ladderPosition
+    setOpponentOptions(
+      [...players]
+        .sort((a, b) => a.ladderPosition - b.ladderPosition)
+        .filter((player) => player.ladderPosition < challengerPosition && player.ladderPosition > challengerPosition - 3 )
```
Comment (adambergman1) [inline components/ReportLadderMatch.js]: For readability reasons, could we maybe place the magic number "3" in a variable?

File: components/ReportLadderMatch.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+      setChallenger(null);
+      setOpponent(null);
+    } catch (error) {
+      console.log(error);
+      setError('Failed to report match. Please try again later.');
+    }
+  };
+
+  return (
+    <div className=''>
```
Comment (adambergman1) [inline components/ReportLadderMatch.js]: Since there is no className provided, we could remove it!

**Review by adambergman1 — COMMENTED** [review]
Really good work! Impressive that this is your first React/JS task! 👏👏👏

File: components/ReportLadderMatch.js (hunk 2 above)
Comment (MikaelZehnterFrisk) [reply components/ReportLadderMatch.js]: Done!

File: components/LadderPlayerItem.js (hunk 1 above)
Comment (MikaelZehnterFrisk) [reply components/LadderPlayerItem.js]: Nice, done!

File: components/ReportLadderMatch.js (hunk 3 above)
Comment (MikaelZehnterFrisk) [reply components/ReportLadderMatch.js]: Very good suggestion!

File: components/ReportLadderMatch.js (hunk 4 above)
Comment (MikaelZehnterFrisk) [reply components/ReportLadderMatch.js]: Done

**Pull request merged**